src/Error/ExceptionTrap.php

```diff
@@ -185,6 +185,83 @@ public function handleException(Throwable $exception): void
         }
     }
 
+    /**
+     * Shutdown handler
+     *
+     * Convert fatal errors into exceptions that we can render.
+     *
+     * @return void
+     */
+    public function handleShutdown(): void
+    {
+        $megabytes = $this->_config['extraFatalErrorMemory'] ?? 4;
```

Comment:
Why isn't this in KB like the helper function?

Reply:
If we change the unit size we'll make upgrading harder as the config keys are shared between the two systems. This has to move anyways as I'm pretty convinced this needs to happen during registration, and at this point it is too late as we could be handling an OOM error and no longer have any space to use for the page rendering process.

Reply:
I was thinking about this more, and our strategy is better if you're able to change memory limits at runtime, and your physical host isn't over-subscribed in memory with no swap.

With our implementation no memory is used up on every single request just in case a fatal error happens. Instead, when an error happens we borrow a bit more RAM to handle the error.